tldw_Server_API/app/core/Chat/chat_service.py

```
@@ -166,6 +168,137 @@ def normalize_request_provider_and_model(
+            # Try reading the raw pricing catalog to preserve original key casing
+            try:
+                cfg_path = Path(__file__).resolve().parents[3] / "Config_Files" / "model_pricing.json"
+                if cfg_path.exists():
+                    data = _json.loads(cfg_path.read_text())
+                    prov_block = data.get(provider) or data.get(provider.capitalize()) or data.get(provider.upper())
+                    if isinstance(prov_block, dict):
+                        return list(prov_block.keys())
+            except (OSError, ValueError, KeyError, AttributeError) as _e:
+                # Expected file/format issues: fall back to normalized catalog
+                logger.debug(f"Model catalog raw load fallback for provider '{provider}': {_e}")
+            except Exception as _ue:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)